graphistry/plugins/spannergraph.py

```diff
@@ -0,0 +1,294 @@
+                                    "destination": element_dict.get('destination_node_identifier'),
+                                    "properties": element_dict.get('properties')
+                                }
+                                json_entry["edges"].append(edge_data)
+                if json_entry["nodes"] or json_entry["edges"]:  # only add non-empty entries
+                    json_list.append(json_entry)
+        return json_list
+
+    @staticmethod
+    def get_nodes_df(json_data: list) -> pd.DataFrame:
+        """
+        Converts spanner json nodes into a pandas DataFrame.
+    
+        :param json_data: The structured JSON data containing graph nodes.
+        :return: A DataFrame containing node information
+        :rtype: pd.DataFrame         
+        """
+        nodes = [
+            { 
+                "label": node.get("label"), 
+                "identifier": node["identifier"], 
+                **node.get("properties", {})
+            }
+            for entry in json_data
+            for node in entry.get("nodes", [])
+        ]
+        nodes_df = pd.DataFrame(nodes).drop_duplicates()
+
+        # if 'type' property exists, skip setting and warn
+        if "type" not in nodes_df.columns:
+            # check 'label' column exists before assigning it to 'type'
+            if "label" in nodes_df.columns:
+                nodes_df['type'] = nodes_df['label']
+            else:
+                nodes_df['type'] = None  # Assign a default value if 'label' is missing
+        else: 
+            logger.warn("unable to assign 'type' from label, column exists\n")
+        
+        return nodes_df
+
+    @staticmethod
+    def get_edges_df(json_data: list) -> pd.DataFrame:
+        """
+        Converts spanner json edges into a pandas DataFrame
+
+        :param json_data: The structured JSON data containing graph edges.
+        :return: A DataFrame containing edge information.
+        :rtype: pd.DataFrame 
+        """
+        edges = [
+            {
+                "label": edge.get("label"),
+                "identifier": edge["identifier"],
+                "source": edge["source"],
+                "destination": edge["destination"],
+                **edge.get("properties", {})
+            }
+            for entry in json_data
+            for edge in entry.get("edges", [])
+        ]
+        edges_df = pd.DataFrame(edges).drop_duplicates()
+
+        # if 'type' property exists, skip setting and warn
+        if "type" not in edges_df.columns:
+            # check 'label' column exists before assigning it to 'type'
+            if "label" in edges_df.columns:
+                edges_df['type'] = edges_df['label']
+            else:
+                edges_df['type'] = None  # Assign a default value if 'label' is missing
+        else: 
+            logger.warn("unable to assign 'type' from label, column exists\n")
+
+        return edges_df
+
+    def gql_to_graph(self, query: str) -> Any:
```

Comment:
more consistent with api to be fluent:

```python
def gql_to_graph(self, g: Plottable, query, ...):
   return g.nodes(...).edges(...)
```

ex:

```
g1 = .... encode_point_colors(...)...settings(...)
g2 = g1.gsql_query(...)
```

Reply:
this one might be a bit more important as it's a public interface (will keep reading)